File: src/main/java/me/itzisonn_/meazy/version/VersionType.java
package me.itzisonn_.meazy.version;

import java.util.Map;

/**
 * Represents different types of version
 */
public enum VersionType {
    /**
     * Alpha version type
     */
    ALPHA,
    /**
     * Beta version type
     */
    BETA,
    /**
     * Release candidate version type
     */
    RELEASE_CANDIDATE,
    /**
     * Release version type
     */
    RELEASE;

    /**
     * @param string String representation of {@link VersionType}
     * @return Version type that matches given string or null
     */
    public static VersionType of(String string) {
        return VERSION_TYPES.get(string.toUpperCase());
    }

    /**
     * Map of different spellings of version types
     */
    public static final Map<String, VersionType> VERSION_TYPES = Map.of(
            "ALPHA", ALPHA,
            "A", ALPHA,
            "BETA", BETA,
            "B", BETA,
            "RELEASE_CANDIDATE", RELEASE_CANDIDATE,
            "RELEASE-CANDIDATE", RELEASE_CANDIDATE,
            "RC", RELEASE_CANDIDATE,
            "RELEASE", RELEASE,
            "R", RELEASE
    );
}

File: src/main/kotlin/me/itzisonn_/meazy/version/VersionType.kt
package me.itzisonn_.meazy.version

import java.util.*

/**
 * Represents different types of version
 */
enum class VersionType {
    /**
     * Alpha version type
     */
    ALPHA,

    /**
     * Beta version type
     */
    BETA,

    /**
     * Release candidate version type
     */
    RELEASE_CANDIDATE,

    /**
     * Release version type
     */
    RELEASE;

    companion object {
        /**
         * @param string String representation of [VersionType]
         * @return Version type that matches given string or null
         */
        fun of(string: String): VersionType? {
            return VERSION_TYPES[string.uppercase(Locale.getDefault())]
        }

        /**
         * Map of different spellings of version types
         */
        val VERSION_TYPES = mapOf(
            "ALPHA" to ALPHA,
            "A" to ALPHA,
            "BETA" to BETA,
            "B" to BETA,
            "RELEASE_CANDIDATE" to RELEASE_CANDIDATE,
            "RELEASE-CANDIDATE" to RELEASE_CANDIDATE,
            "RC" to RELEASE_CANDIDATE,
            "RELEASE" to RELEASE,
            "R" to RELEASE
        )
    }
}
